stablehlo/dialect/StablehloOps.cpp

```diff
@@ -4024,46 +4029,19 @@ LogicalResult ReduceOp::verify() {
     }
   }
 
-  Block& block = body().front();
+  Block& block = adaptor.body().front();
   SmallVector<TensorType> accumulatorSubShapes;
-  if (failed(verifyReducerShape(this->getLoc(), block, inputArgTypes,
+  if (failed(verifyReducerShape(location.getValue(), block, inputArgTypes,
```

Comment:
This is not going to work well if location is empty. You'll need to change `verifyReducerShape` to take an optional location.

Reply:
This is an issue indeed. Change `verifyReducerShape` to accept `Optional<Location>`